### DISCUSSION TITLE
Status bar as opposed to progress bar?

### DISCUSSION THREAD

**Discussion by rb-synth** [Textualize/rich#3279] [Q&A]
A progress bar implies a binary state – a chunk of work is either done or not yet done. Can I have, say, 3 states? I have a cluster node that is either being used, sleeping or free. I have my own implementation that works, but want to simplify by using `rich`. Can I have the same implementation? 

In the given example, the number of 🔺's represents 1/16th of the bar width.

g5-queue1 |🔺🔺🔺🔺🔺💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤💤🌱🌱🌱🌱🌱🌱🌱🌱🌱🌱🌱🌱🌱🌱🌱| 🔺: 1    💤: 12   🌱: 3    Tot: 16

**Comment by peterjc** [comment]
I was wondering this too, for jobs submitted to a cluster which have 3 or four states:
* Pending
* Running
* Complete (or error)

A progress bar with multiple colours _might_ work - essentially expanding the current active "point" splitting done and pending (two colours, default red and black), into a variable length mid-section for running jobs (so three colours). e.g.

```
Computing ━━━━━╺━━━╺━━━━━━━━━━━━━━━━━━━━━━━━   6% 0:00:36
```

**Reply by rb-synth** [reply]
Hey @peterjc I guess what you're looking for is slightly different from what I'm looking for. You're still looking for a progress bar, but one that has a little more information attached (the uncompleted work is split into running and pending), whereas I'm looking for a status bar. In my example, I expect the three bars to vary over time, but there is no end goal.  

Perhaps the same solution will work for both of us though, of course.
